lib/logflare/backends/adaptor/clickhouse_adaptor/pipeline.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Logflare.Backends.Adaptor.ClickhouseAdaptor.Pipeline do
+  @moduledoc """
+  Pipeline for `ClickhouseAdaptor`
+
+  This pipeline is responsible for taking log events from the
+  source backend and inserting them into the configured database.
+  """
+
+  alias Broadway.Message
+  alias Logflare.Backends.Adaptor.ClickhouseAdaptor
+  alias Logflare.Backends.BufferProducer
+
+  @doc false
+  @spec start_link(ClickhouseAdaptor.t()) ::
+          {:ok, pid()} | :ignore | {:error, {:already_started, pid()} | term()}
+  def start_link(%ClickhouseAdaptor{} = adaptor_state) do
+    Broadway.start_link(__MODULE__,
+      name: adaptor_state.pipeline_name,
+      producer: [
+        module:
+          {BufferProducer,
+           [source_id: adaptor_state.source.id, backend_id: adaptor_state.backend.id]},
+        transformer: {__MODULE__, :transform, []},
+        concurrency: 1
+      ],
+      processors: [
+        default: [concurrency: 5, min_demand: 1]
+      ],
+      batchers: [
+        ch: [concurrency: 5, batch_size: 350]
```

Comment:
Probably going to need to be able configure the batch size but we can punt this for a bit

Reply:
Sounds good. I did switch them to module attributes so they are easy to spot/change in the meantime.